### PR TITLE
.Net: Avoid causing first-chance exceptions in CodeTokenizer

### PR DESCRIPTION
**Pull request by glahaye**
### Motivation and Context
The CodeTokenizer routinely causes first-chance (caught) exceptions to determine whether tokens belong to function names of named argument blocks. This is inefficient but also can be startling depending on how one has set one's configuration for breaking on exceptions.

See #4418 

### Description
CodeTokenizer now uses a new method from NamedArgBlock to determine whether a string could result in a NamedArgBlock instead of resorting to causing exceptions to gather the same information.

### Contribution Checklist
- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Review by crickman — APPROVED** [review]

File: dotnet/src/SemanticKernel.Core/TemplateEngine/Blocks/NamedArgBlock.cs [hunk 1]
```diff
@@ -61,6 +56,22 @@ public NamedArgBlock(string? text, ILoggerFactory? logger = null)
...
     }
 
+    /// <summary>
+    /// Attempts to extract the name and value of a named argument block from a string
+    /// </summary>
+    /// <param name="text">String from which to extract a name and value</param>
+    /// <param name="argParts">This array has its first and second elements set to the extracted name and value when successful.
+    /// An empty array is returned otherwise.</param>
+    /// <returns>true when a name and value are successfully extracted from the given text</returns>
+    internal static bool TryGetNameAndValue(string? text, out string[] argParts)
```
Comment (markwallace-microsoft) [inline dotnet/src/SemanticKernel.Core/TemplateEngine/Blocks/NamedArgBlock.cs:66]: Consider defining this as `bool TryGetNameAndValue(string? text, out string name, out string value)`

File: dotnet/src/SemanticKernel.Core/TemplateEngine/Blocks/NamedArgBlock.cs [hunk 2]
```diff
@@ -61,6 +56,22 @@ public NamedArgBlock(string? text, ILoggerFactory? logger = null)
...
+    {
+        text ??= string.Empty;
```
Comment (markwallace-microsoft) [inline dotnet/src/SemanticKernel.Core/TemplateEngine/Blocks/NamedArgBlock.cs:68]: If text is null or empty should just return false

**Review by markwallace-microsoft — COMMENTED** [review]
Few minor comments

File: dotnet/src/SemanticKernel.Core/TemplateEngine/Blocks/NamedArgBlock.cs (hunk 1 above)
Comment (glahaye) [reply dotnet/src/SemanticKernel.Core/TemplateEngine/Blocks/NamedArgBlock.cs:66]: I like it

File: dotnet/src/SemanticKernel.Core/TemplateEngine/Blocks/NamedArgBlock.cs (hunk 2 above)
Comment (glahaye) [reply dotnet/src/SemanticKernel.Core/TemplateEngine/Blocks/NamedArgBlock.cs:68]: I like this too!

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**